CMakeLists.txt

```
@@ -30,60 +34,115 @@ include(CMakePackageConfigHelpers)
 
 option(BUILD_END2END "Build End2End test" OFF)
 option(BUILD_TESTS "Build Unit Tests" OFF)
+option(BUILD_COMMON "Build Common" ON)
 option(BUILD_WORKER "Build Worker" ON)
 option(BUILD_CLIENT "Build Client" ON)
 option(BUILD_DYNAMICWORKER "Build DynamicWorker" OFF)
+option(BUILD_WORKERTEST "Build Worker Test" OFF)
+option(BUILD_SDK "Build SDK" ON)
 
-function(setup_options project_name_param)
+include(ExternalProject)
+# if(NOT BUILD_WORKERTEST)
+# ExternalProject_Add(ArmoniK.SDK.Common
+#     SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/ArmoniK.SDK.Common"
+#     BINARY_DIR "${CMAKE_CURRENT_SOURCE_DIR}/build/ArmoniK.SDK.Common"
+#     CMAKE_ARGS -DCMAKE_INSTALL_PREFIX:PATH=${INSTALL_SDK_DIR} -DCMAKE_PREFIX_PATH=${ARMONIK_API_DIR}
+#     BUILD_ALWAYS TRUE
+#     STEP_TARGETS build
+# )
+# endif()
 
-    file(READ /etc/issue ${PROJECT_NAME}_ETCISSUE_CONTENT)
-    string(FIND "${${project_name_param}_ETCISSUE_CONTENT}" "Alpine" IS_ALPINE)
+# if(BUILD_WORKER AND NOT BUILD_WORKERTEST)
+#     ExternalProject_Add(ArmoniK.SDK.Worker
+#         SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/ArmoniK.SDK.Worker"
+#         BINARY_DIR "${CMAKE_CURRENT_SOURCE_DIR}/build/ArmoniK.SDK.Worker"
+#         CMAKE_ARGS -DCMAKE_INSTALL_PREFIX:PATH=${INSTALL_SDK_DIR} -DCMAKE_PREFIX_PATH=${ARMONIK_API_DIR}
+#         DEPENDS ArmoniK.SDK.Common
+#         BUILD_ALWAYS TRUE
+#         STEP_TARGETS builds
+#     )
+# endif()
 
-    if(${ARGC} GREATER 1)
-        set(extra_param ${ARGV1})
-    endif()
+include_directories(/app/install/include)
+if(BUILD_SDK)
+    ExternalProject_Add(ArmoniK.SDK.Common
+        SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/ArmoniK.SDK.Common"
+        BINARY_DIR "${CMAKE_CURRENT_SOURCE_DIR}/build/ArmoniK.SDK.Common"
+        CMAKE_ARGS -DCMAKE_INSTALL_PREFIX:PATH=${INSTALL_SDK_DIR} -DCMAKE_PREFIX_PATH=${ARMONIK_API_DIR}
+        BUILD_ALWAYS TRUE
+        STEP_TARGETS build
+    )
 
-    if(MSVC)
-        target_compile_options(${project_name_param} PRIVATE /W4)
-    else()
-        if(CMAKE_BUILD_TYPE MATCHES DEBUG AND IS_ALPINE EQUAL -1)
-            target_compile_options(${project_name_param} PRIVATE -Wall -Wextra -Wpedantic -fsanitize=undefined,address ${extra_param})
-        else ()
-            target_compile_options(${project_name_param} PRIVATE -Wall -Wextra -Wpedantic ${extra_param})
-        endif()
+    if(BUILD_WORKER)
+        ExternalProject_Add(ArmoniK.SDK.Worker
+            SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/ArmoniK.SDK.Worker"
+            BINARY_DIR "${CMAKE_CURRENT_SOURCE_DIR}/build/ArmoniK.SDK.Worker"
+            CMAKE_ARGS -DCMAKE_INSTALL_PREFIX:PATH=${INSTALL_SDK_DIR} -DCMAKE_PREFIX_PATH=${ARMONIK_API_DIR}
+            DEPENDS ArmoniK.SDK.Common
```

```suggestion
            DEPENDS ArmoniK.SDK.Common
```
Actually doesn't depend on it